certchain/certmanager/multicert_disk_cache.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certmanager
+
+import (
+	"bytes"
+	"errors"
+	"path"
+
+	"github.com/gofrs/flock"
+	"github.com/google/renameio"
+	"github.com/google/webpackager/certchain"
+	"github.com/google/webpackager/certchain/certchainutil"
+	"github.com/hashicorp/go-multierror"
+)
+
+var (
+	errEmptyCertDir      = errors.New("certmanager: empty CertDir")
+	errEmptyCertFilename = errors.New("certmanager: empty CertFilename")
+	errEmptyOCSPFilename = errors.New("certmanager: empty OCSPFilename")
+)
+
+// MultiCertDiskCache is a Cache on a local filesystem. It writes the
+// certificate chain in the PEM format and the OCSP response in the DER format
+// to separate files as specified by MultiCertDiskCacheConfig. It uses the
+// digest of the certificate chain as the base filename for the cert and OCSP
+// files.
+type MultiCertDiskCache struct {
+	MultiCertDiskCacheConfig
+}
+
+var _ Cache = (*MultiCertDiskCache)(nil)
+
+// MultiCertDiskCacheConfig configures DiskCache.
+type MultiCertDiskCacheConfig struct {
+	// CertDir locates the directory to write the certificate chain to.
+	// If CertDir is empty, the certificate chain is not cached.
```

Comment:
I think we can require CertDir to be always nonempty. Maybe same for other fields in this struct.

Reply:
Done